pkg/types/validation/installconfig.go

```diff
@@ -193,14 +191,8 @@ func validateCompute(platform *types.Platform, pools []types.MachinePool, fldPat
 			allErrs = append(allErrs, field.Duplicate(poolFldPath.Child("name"), p.Name))
 		}
 		poolNames[p.Name] = true
-		if p.Replicas != nil && *p.Replicas > 0 {
-			foundPositiveReplicas = true
-		}
 		allErrs = append(allErrs, ValidateMachinePool(platform, &p, poolFldPath)...)
 	}
-	if !foundPositiveReplicas {
-		logrus.Warnf("There are no compute nodes specified. The cluster will not fully initialize without compute nodes.")
```

Comment:
Can we really remove this warning?  Does this PR allow the ingress scheduler to be scheduled on control plane nodes (e.g. see [here][1] and kubernetes/kubernetes#65618)?  Are we running CI on zero-compute clusters?

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1671136#c1

Reply:
Yes, when you set schedulableMasters to true, the worker role gets added to the master nodes, as well.

nothing in OpenShift CI for this yet

Reply:
On cloud platforms, while adding a worker role to masters will allow ingress controllers to be scheduled on the masters, the master nodes will remain excluded from load balancer target pools, breaking the practical feature.

As to whether Kube will support including masters in LB target pools, the jury is still out and there is still no KEP AFAICT.

https://github.com/kubernetes/kubernetes/issues/65618
https://groups.google.com/d/msg/kubernetes-sig-network/erULVdDkLNw/hvCukZ3CBQAJ

Reply:
@russelb, can you restore the warning?

Reply:
Also, with a single compute, we won't trigger this schedulable-control-plane logic, so won't ingress hang with "I can't get replicas up to two"?

Reply:
The warning wouldn't be appropriate for a 3-node bare metal cluster.  I guess it could be restored and only emitted if the platform type is not `baremetal` or `none` because of the load balancer issue mentioned by @ironcladlou ?